apps/rush-lib/src/logic/StandardScriptUpdater.ts

```
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as path from 'path';
+import { Text } from '@microsoft/node-core-library';
+
+import { RushConfiguration } from '../api/RushConfiguration';
+
+/**
+ * Checks whether the common/scripts files are up to date, and recopies them if needed.
+ * This is used by the "rush install" and "rush update" commands.
+ */
+export class StandardScriptUpdater {
+  private static readonly _scriptNames: string[] = [ 'install-run.js', 'install-run-rush.js' ];
+
+  /**
+   * Recopy the scripts if the scripts are out of date.
+   * Used by "rush update".
+   */
+  public static update(rushConfiguration: RushConfiguration): boolean {
+    let anyChanges: boolean = false;
+    for (const scriptName of StandardScriptUpdater._scriptNames) {
+      if (StandardScriptUpdater._updateScript(scriptName, rushConfiguration, false)) {
+        anyChanges = true;
+      }
+    }
+
+    if (anyChanges) {
+      console.log();
+    }
+    return anyChanges;
+  }
+
+  /**
+   * Throw an exceptions if the scripts are out of date.
+   * Used by "rush install".
+   */
+  public static validate(rushConfiguration: RushConfiguration): void {
```

>validate [](start = 16, length = 8)

It seems like this function should throw the exception.